flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/resource/UserResourceManagerTest.java

```diff
@@ -0,0 +1,195 @@
+        try {
+            Class.forName(LOWER_UDF_CLASS, false, userClassLoader);
+            fail("Should fail.");
+        } catch (ClassNotFoundException e) {
+        }
+
+        // register the same jar repeatedly
+        userResourceManager.registerResource(new ResourceUri(ResourceType.JAR, udfJar.getPath()));
+        userResourceManager.registerResource(new ResourceUri(ResourceType.JAR, udfJar.getPath()));
+
+        // assert resource infos
+        Map<ResourceUri, URL> expected =
+                Collections.singletonMap(
+                        new ResourceUri(ResourceType.JAR, udfJar.getPath()),
+                        userResourceManager.getURLFromPath(new Path(udfJar.getPath())));
+
+        assertEquals(expected, userResourceManager.getResources());
+
+        // test load class
+        final Class<?> clazz1 = Class.forName(LOWER_UDF_CLASS, false, userClassLoader);
+        final Class<?> clazz2 = Class.forName(LOWER_UDF_CLASS, false, userClassLoader);
+
+        assertEquals(clazz1, clazz2);
+    }
+
+    @Test
+    public void testRegisterInvalidResource() throws Exception {
+        UserResourceManager userResourceManager = createResourceManager(new URL[0]);
+
+        // test register non-exist file
+        String uri = temporaryFolder.getRoot().getPath() + Path.SEPARATOR + "test-file";
+        try {
+            userResourceManager.registerResource(new ResourceUri(ResourceType.FILE, uri));
+            fail("Should fail.");
+        } catch (IllegalArgumentException e) {
+            assertThat(e)
+                    .satisfies(
+                            matching(
+                                    containsMessage(
+                                            String.format("Resource [%s] not found.", uri))));
+        }
+
+        // test register directory for jar resource
+        uri = temporaryFolder.newFolder("test-jar-dir").getPath();
+        try {
+            userResourceManager.registerResource(new ResourceUri(ResourceType.JAR, uri));
+            fail("Should fail.");
+        } catch (IllegalArgumentException e) {
+            assertThat(e)
+                    .satisfies(
+                            matching(
+                                    containsMessage(
+                                            String.format(
+                                                    "Directory [%s] is not allowed for registering jar resource.",
+                                                    uri))));
+        }
+    }
+
+    @Test
+    public void testDownloadResource() throws Exception {
+        Path srcPath = new Path(udfJar.getPath());
+        UserResourceManager userResourceManager = createResourceManager(new URL[0]);
+
+        // test download resource to local path
+        URL localUrl = userResourceManager.downloadResource(srcPath);
+
+        String expected = FileUtils.readFileUtf8(udfJar);
+        String actual = FileUtils.readFileUtf8(new File(localUrl.getFile()));
```

Comment:
Would be better to use `org.apache.flink.util.FileUtils#readAllBytes`. The jar file may not be compiliant with UTF-8 encoding.